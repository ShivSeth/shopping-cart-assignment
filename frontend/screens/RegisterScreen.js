import React, { useState } from 'react';
import { useHistory } from 'react-router';

import './RegisterScreen.scss';

const RegisterScreen = () => {
  const [errorMsg, setError] = useState('');
  const [input, setInput] = useState({
    firstName: '',
    lastName: '',
    email: '',
    password: '',
    confirmPassword: '',
  });
  const history = useHistory();

  function formSubmit(e) {
    e.preventDefault();
    console.log(input);
    history;

    if (input.password !== input.confirmPassword) {
      setError("Invalid Input or Passwords didn't match");
      setTimeout(() => {
        setError('');
      }, 3000);
    } else {
      history.push('/');
    }
  }

  const handleChange = (e) => {
    setInput((values) => ({
      ...values,
      [e.target.name]: e.target.value,
    }));
  };

  return (
    <div className="login-form">
      <aside className="left-side">
        <h2>Sign Up</h2>
        <p>We do not share your personal details with anyone</p>
      </aside>
      <form className="right-side" onSubmit={formSubmit}>
        <input
          value={input.firstName}
          required
          onChange={handleChange}
          type="text"
          name="firstName"
          placeholder="First Name"
        />
        <input
          value={input.lastName}
          onChange={handleChange}
          required
          type="text"
          name="lastName"
          placeholder="Last Name"
        />
        <input
          value={input.email}
          onChange={handleChange}
          required
          type="email"
          name="email"
          placeholder="Email"
        />
        <input
          value={input.password}
          onChange={handleChange}
          required
          type="password"
          name="password"
          pattern="(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{6,}"
          minLength="6"
          placeholder="Password"
        />
        <input
          value={input.confirmPassword}
          onChange={handleChange}
          required
          type="password"
          name="confirmPassword"
          pattern="(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{6,}"
          minLength="6"
          placeholder="Confirm Password"
        />
        {errorMsg && <p className="error-msg">{errorMsg}</p>}
        <button type="submit">Sign In</button>
      </form>
    </div>
  );
};

export default RegisterScreen;
