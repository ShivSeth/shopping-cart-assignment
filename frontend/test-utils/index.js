import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { AuthContextProvider } from '../context/AuthContext';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import store from '../store';
store;
const AllTheProviders = ({ children }) => {
  const history = createMemoryHistory();

  return (
    <Provider store={store}>
      <AuthContextProvider>
        <Router history={history}>{children}</Router>
      </AuthContextProvider>
    </Provider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
