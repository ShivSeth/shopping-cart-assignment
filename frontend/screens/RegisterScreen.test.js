import { render, screen } from '../test-utils';
import RegisterScreen from './RegisterScreen';

describe('<LoginScreen/>', () => {
  test('should sign in input boxes be present ', () => {
    const { container } = render(<RegisterScreen />);
    const firstNameInput = screen.getByPlaceholderText('First Name');
    const lastNameInput = screen.getByPlaceholderText('Last Name');
    const emailInput = screen.getByPlaceholderText('Email');
    const passwordInput = screen.getByPlaceholderText('Password');
    const confirmPasswordInput =
      screen.getByPlaceholderText('Confirm Password');
    expect(firstNameInput).toBeInTheDocument();
    expect(lastNameInput).toBeInTheDocument();
    expect(emailInput).toBeInTheDocument();
    expect(passwordInput).toBeInTheDocument();
    expect(confirmPasswordInput).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
