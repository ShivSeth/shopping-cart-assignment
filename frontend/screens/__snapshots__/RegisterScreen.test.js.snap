// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginScreen/> should sign in input boxes be present  1`] = `
<div>
  <div
    class="login-form"
  >
    <aside
      class="left-side"
    >
      <h2>
        Sign Up
      </h2>
      <p>
        We do not share your personal details with anyone
      </p>
    </aside>
    <form
      class="right-side"
    >
      <input
        name="firstName"
        placeholder="First Name"
        required=""
        type="text"
        value=""
      />
      <input
        name="lastName"
        placeholder="Last Name"
        required=""
        type="text"
        value=""
      />
      <input
        name="email"
        placeholder="Email"
        required=""
        type="email"
        value=""
      />
      <input
        minlength="6"
        name="password"
        pattern="(?=.*\\\\d)(?=.*[a-z])(?=.*[A-Z]).{6,}"
        placeholder="Password"
        required=""
        type="password"
        value=""
      />
      <input
        minlength="6"
        name="confirmPassword"
        pattern="(?=.*\\\\d)(?=.*[a-z])(?=.*[A-Z]).{6,}"
        placeholder="Confirm Password"
        required=""
        type="password"
        value=""
      />
      
      <button
        type="submit"
      >
        Sign In
      </button>
    </form>
  </div>
</div>
`;
